### ISSUE TITLE
Grabbing frames from the PipeWire stream isn't explained.

### ISSUE THREAD

**Issue by Frostie314159** [bilelmoussaoui/ashpd#155]
There is no documentation on how to actually read frames from the PipeWire stream acquired with screencast.

**Comment by bilelmoussaoui** [comment]
Feel free to submit a patch. I am not a pipewire hacker, I only pass the fd & node_id to gstreamer :)

**Comment by A6GibKm** [comment]
> ashpd-demo isn't very clear on this.

Is basically *the* example.

**Comment by bilelmoussaoui** [comment]
Have you checked https://github.com/bilelmoussaoui/ashpd/blob/master/ashpd-demo/src/widgets/gst_paintable.rs? there is no "small" example as far as i know, feel free to build one :)

**Comment by SyedAhkam** [comment]
Thanks for the prompt response. Yes. I had checked out `gst_paintable.rs` but as I said it wasn't too clear mixed with UI logic and believe me I tried my hardest to make this work the past few days, either with gstreamer or pipewire (but honestly more with pipewire)

Finally got some success right after I commented here - so basically I ended up mixing this example from [pipewire-rs](https://gitlab.freedesktop.org/pipewire/pipewire-rs/-/blob/86df39190c0ab67444666a42908f7e8c1344e24a/pipewire/examples/streams.rs) into the docs.rs example for ashpd on `Screencast` and came up with [this](https://gist.github.com/SyedAhkam/6131487ea9caaf6113426ecfb7104a75).

Although my attempt was non-gstreamer based, I think it solves the crux of this issue. If it looks good to you guys, I can go ahead and make a PR to include this as a standalone example?
